lib/ably/realtime/client/incoming_message_dispatcher.rb

```diff
@@ -102,7 +102,11 @@ def dispatch_protocol_message(*args)
               if channel.attached?
                 channel.manager.duplicate_attached_received protocol_message
               else
-                channel.transition_state_machine :attached, reason: protocol_message.error, resumed: protocol_message.has_channel_resumed_flag?, protocol_message: protocol_message
+                if channel.failed?
+                  logger.warn "Ably::Realtime::Client::IncomingMessageDispatcher - Received an ATTACHED protocol message for FAILED channel #{channel.name}. Ignoring ATTACHED message"
```

Comment:
Is there a spec for this? Having the client deliberately ignore what the server is telling it the channel state is is a bit of a break from our usual model (of the server deciding and telling the client), no?

Reply:
This wasn't a change of behaviour.  This was raising an exception before because a channel is not allowed to transition from `failed` directly to `attached`.  See https://github.com/ably/ably-ruby/blob/f7a524e00e64a47cf5c9d974dbcf621f5b82307a/lib/ably/realtime/channel/channel_state_machine.rb#L29.  I perhaps should investigate more under what conditions this has happened (I saw this exception bubble up in one of the tests), but I'd prefer to create a separate issue as it's not related to this PR, this change just made the library handle the problem better.